sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExec.scala

```diff
@@ -1636,15 +1641,18 @@ class GpuRunningWindowIterator(
     ret
   }
 
-  override def hasNext: Boolean = {
+  override def hasNext: Boolean = maybeSplitIter.hasNext || {
     cacheBatchIfNeeded()
     cachedBatch.isDefined
   }
 
   override def next(): ColumnarBatch = {
-    val cb = readNextInputBatch()
+    if (!maybeSplitIter.hasNext) {
+      maybeSplitIter = computeRunningAndClose(readNextInputBatch())
+      assert(maybeSplitIter.hasNext)
```

Comment:
nit: Let's avoid using assert here.

Reply:
Thanks for catching this.
If this is a nit, i will keep it in this PR to avoid another premerge run.

Reply:
Removed it.